src/seedu/addressbook/data/person/Address.java

```diff
@@ -26,6 +31,11 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(trimmedAddress)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
+        String[] addressParts = address.split(", ");
+        block = new Block(addressParts[0]);
+        street = new Street(addressParts[1]);
+        unit = new Unit(addressParts[2]);
+        postalCode = new PostalCode(addressParts[3]);
```

Comment:
Why are you still retaining the `value`? Think about the havoc data duplication can cause.